commit 9e4a617757273a86b560c1ece40c48e4940a3c79
Author: Kees Cook <kees@kernel.org>
Date:   Thu Sep 29 02:24:53 2022 -0700

    string: Add __realloc_size hint to kmemdup()
    
    Add __realloc_size() hint to kmemdup() so the compiler can reason about
    the length of the returned buffer. (These must not use __alloc_size,
    since those include __malloc which says the contents aren't defined[1]).
    
    [1] https://lore.kernel.org/linux-hardening/d199c2af-06af-8a50-a6a1-00eefa0b67b4@prevas.dk/
    
    Cc: Rasmus Villemoes <rasmus.villemoes@prevas.dk>
    Cc: Guenter Roeck <linux@roeck-us.net>
    Cc: Andy Shevchenko <andriy.shevchenko@intel.com>
    Cc: Paolo Abeni <pabeni@redhat.com>
    Cc: Geert Uytterhoeven <geert@linux-m68k.org>
    Signed-off-by: Kees Cook <keescook@chromium.org>

diff --git a/include/linux/string.h b/include/linux/string.h
index cf7607b32102..db28802ab0a6 100644
--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -176,7 +176,7 @@ extern void kfree_const(const void *x);
 extern char *kstrdup(const char *s, gfp_t gfp) __malloc;
 extern const char *kstrdup_const(const char *s, gfp_t gfp);
 extern char *kstrndup(const char *s, size_t len, gfp_t gfp);
-extern void *kmemdup(const void *src, size_t len, gfp_t gfp);
+extern void *kmemdup(const void *src, size_t len, gfp_t gfp) __realloc_size(2);
 extern char *kmemdup_nul(const char *s, size_t len, gfp_t gfp);
 
 extern char **argv_split(gfp_t gfp, const char *str, int *argcp);